test/test_preprocessing_encoders.py

```diff
@@ -578,3 +579,32 @@ def test_woe_multi_cols():
     Xe = enc.fit_transform(X, titanic_y)
     assert len(np.unique(Xe[:, 0])) == 4
     assert len(np.unique(Xe[:, 1])) == 4
+
+
+def test_similarity_consistent():
```

Comment:
Can you also check whether the more similar the strings, the more similar the encoded vectors? Say, the distance between "cat1" and "cat2" should be less than between "cat1" and "dog".

Reply:
I added some additional tests to this function checking exactly this.